helper/include/odp/helper/threads.h

```diff
@@ -191,6 +191,33 @@ typedef struct {
 
 } odph_thread_common_param_t;
 
+/** Thread join result type */
+typedef struct {
+	/** Variant field mappings for thread/process modes */
+	union {
```

Comment:
The usage of this struct seems quite complex in the later commits. Would it be possible to forgo this union and use the same `ret` for both threads and processes?  So it would be enough for an application to check `odph_thread_join_res_t.ret != 0`? Assuming there is no signal with value zero.

Reply:
It follows the same style as launching threads. In any case, not having some indication about the signaled condition would make the return value a bit useless in debugging.

Reply:
The same information would still be provided.
```
/** Thread join result */
typedef struct {
	/**
	 * Return value of the exited thread/process
	 *
	 * If 'is_sig' is true, then this is the signal number
	 * that caused the process exit. Otherwise exit status of the
	 * thread/process.
	 */
	int ret;

	/** Exit caused by signal */
	odp_bool_t is_sig;

} odph_thread_join_result_t;
```